src/main/java/org/broadinstitute/hellbender/utils/hmm/interfaces/AlleleMetadataProvider.java

```diff
@@ -0,0 +1,26 @@
+package org.broadinstitute.hellbender.utils.hmm.interfaces;
+
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.vcf.VCFHeader;
+
+import javax.annotation.Nonnull;
+
+/**
+ * An interface for hidden states which are construable as {@link Allele}.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public interface AlleleMetadataProvider {
```

Comment:
`Provider` generally implies that this would be a factory that dynamically selects an instance of an abstract class i.e. `AlleleMetadata`.  Since this is not the intent of this code you should rename the interface to something that conveys that an implementing class or enum *has* allele metadata.